### ISSUE TITLE
Ternary conditions on Arrays/Matrices

### ISSUE THREAD

**Issue by SimoneTosato** [josdejong/mathjs#718]
When I try something like 

```
a=-5:5
a<0
[true, true, true, true, true, false, false, false, false, false, false]
```
everything works fine, but with ternary conditions

```
(a<0)?1:0
TypeError: Unsupported type of condition "Matrix"
```

isn't this supposed to return something like this?

```
[ 1 , 1 , 1 , 1 , 1 , 0 , 0 , 0 , 0 , 0 ]
```

**Comment by josdejong** [comment]
The conditional operator indeed doesn't support element wise operations. The reason is that you might want different behaviors, like should `[true, false] ? [1,2] : [3,4]`return `[3,2]` or `[[1,2], [3,4]]`?

For your use case, you can use the function `map` like:
```
a=-5:5
    [-5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5]
a<0
    [true, true, true, true, true, false, false, false, false, false, false]
map(a, f(x) = x < 0 ? 1 : 0)
    [1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0]
```

**Comment by SimoneTosato** [comment]
Any plan to implement an element wise version of the conditional operator? It could be useful to have a quick way other than the function to map an array of booleans with values, maybe with the usual element wise syntax

```
a=-5:5
a<0 .? 1 : 0
    [1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0]
a<0 .? [1,2] : [3,4]
    [[1,2], [1,2], [1,2], [1,2], [1,2], [3,4], [3,4], [3,4], [3,4], [3,4], [3,4]]
```

**Comment by josdejong** [comment]
Yes, I think it makes sense to re-evaluate this. In a very old version of mathjs, element wise operations on the conditional operator was returning the matching case also element wise, which was confusing.

Interesting idea to introduce a new operator `.?`. I think we can just extend the existing operator `?` though since there it doesn't support element wise operation so there would be no conflict. It should behave like:
```js
[true, false, true] ? [1,2] : [3,4]
    [[1,2], [3,4], [1,2]]
```

Optionally we could introduce a `.?` operator which could have behavior like this:
```js
[true, false, true] ? [1,2,3] : [4,5,6]
    [1, 5, 3]
```
But I'm not sure if there is much need for this, and it can be done using `map` too so it's sort of redundant:
```js
a=[1,2,3]
b=[4,5,6]
map([true, false, true], f(x,i) = x ? a[i[1]] : b[i[1]])
```

**Comment by SimoneTosato** [comment]
My idea to use `.?` was to be more explicit to about the element wise nature of the operator but yeah i don't think it's really necessary to use the dot notation

**Comment by josdejong** [comment]
The only edge case would be when you use the conditional operator to see whether a variable contains "something" and if so do A, else do B. But I think the more common case is the one we're discussing now, and if you want to test whether a variable contains "something" you can explicitly test via `myVar == null ? ... : ...`.

**Comment by ddavidebor** [comment]
for our (mine and @SimoneTosato) application our need @josdejong is the one you indicated in your pointwise example:

```js
[true, false, true] ? [1,2,3] : [4,5,6]
     [1, 5, 3]
```

**Comment by josdejong** [comment]
@ddavidebor thanks  for the feedback, good to hear that there are practical use cases. So we could go for my proposal at https://github.com/josdejong/mathjs/issues/718#issuecomment-250968192, implement a new operator `.?` for this behavior.

**Comment by ddavidebor** [comment]
@josdejong yes please that would enable the usage of thernary conditions in our application

**Comment by josdejong** [comment]
@ddavidebor the `.?` operator would be convenient indeed, but don't see this as a showstopper: you could always create and import your own functions to mathjs like:

```js
math.import({
    iff: function (condition, truePart, falsePart) {
      // ... element wise implementation of conditional operator ...
    }
})

math.eval('iff([true, false, true] ? [1,2,3] : [4,5,6])');  // [1, 5, 3]
```

**Comment by ddavidebor** [comment]
@josdejong yes of course we can work around that, but it's not as elegant and practical for the end user of something that works with the equation parser.

**Comment by josdejong** [comment]
agree
